compiler/rustc_resolve/src/late.rs

```diff
@@ -1949,6 +1982,7 @@ impl<'a: 'ast, 'b, 'ast> LateResolutionVisitor<'a, 'b, 'ast> {
                 )
                 .span_label(param.ident.span, "'static is a reserved lifetime name")
                 .emit();
+                // Do not record lifetime res, so lowering knows there is something fishy.
```

Comment:
Maybe it's better to use `LifetimeRes::Error` if lifetime resolutions are recorded in all other cases?